src/js/actions/search/adobestock.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+define(function (require, exports) {
+    "use strict";
+
+    var Immutable = require("immutable");
+
+    var events = require("js/events"),
+        locks = require("js/locks");
+
+    var searchActions = require("js/actions/search");
+    
+    /**
+     * Captures search term for Adobe Stock and executes search
+     *
+     * @private
+     * @param {string} searchTerm
+    */
+    var _confirmStockSearch = function (searchTerm) {
+        searchActions.executeStockSearch.call(this, searchTerm);
+    };
+
+    /**
+     * Find SVG class for Adobe Stock
+     * If this needs to vary based on the item, use category list as parameter 
+     * (see getSVGCallback type in search store)
+     * 
+     * @return {string}
+    */
+    var _getSVGClass = function () {
+        return "libraries-stock";
+    };
+
+    /**
+     * Make list for Stock options, currently empty
+     * 
+     * @private
+     * @return {Immutable.List.<object>}
+    */
+    var _adobeStockSearchOptions = function () {
```

Comment:
Oh, I see what's happening now. But, I suggest just using an anonymous function literal in `registerAdobeStock` instead of a named, documented function as above. Also, it doesn't need to be bound with `.bind` because the body doesn't refer to `this`. (The only thing `.bind` does is set the value of `this`.)